components/compilers/typescript/index.js

```diff
@@ -1,66 +1,127 @@
+        jsx: 'react',
+        target: 'es5',
+        incremental: true,
+        isolatedModules: true
+    
+    };
+
+    // Divide files by whether we should compile them, according to file type.
+    const filesByToCompile = groupBy(files, _toCompile, context);
+
+    // const compiled = (!filesByToCompile.true || filesByToCompile.true.length === 0) ? [] : filesByToCompile.true.map(file => compileSingleFile(file, compilerOptions, distPath)).reduce((a, b) => a.concat(b));
+    // console.time('compileComponent')
+    const compiled = (!filesByToCompile.true || filesByToCompile.true.length === 0) ? [] : compileComponent(filesByToCompile.true, compilerOptions, distPath, context)
+    // console.timeEnd('compileComponent')
+    const nonCompiled = !filesByToCompile.false ? [] : filesByToCompile.false.map(file => _getDistFile(file, distPath, false));
+    // console.timeEnd('compile')
+    return compiled.concat(nonCompiled);
 }
 
+let oldProgram = undefined
+export function compileComponent(files, options, path, context) {
+    // console.time('create')
+    let program = ts.createProgram(files.map(file => file.path), options, undefined, oldProgram);
+    oldProgram = program
+    // console.timeEnd('create')
+    let declarationFile;
+    const sources = program.getSourceFile(context.componentObject.mainFile)
+    // console.time('emit')
+    const emitResults = program.emit(sources, (file, data) => {
+        declarationFile = new Vinyl({
+            contents: new Buffer(data),
+            base: path,
+            path: _getDistFilePath({relative:file.split(longestStartingSubstring(file, path))[1]}, path, true).replace('d.js', 'd.ts'),
+        });
+    },undefined, true);
+    // console.timeEnd('emit')
+
+    const results = files.map(file => {
+      if (isDefJS(file)) {
+        return null
+      }
+        const result = ts.transpileModule(file.contents.toString(), 
+            { 
+                compilerOptions: options, 
+                fileName: file.basename, 
+                moduleName: file.basename.split('.')[0]
+            });
+        const mappings = new Vinyl({
+            contents: new Buffer(result.sourceMapText),
+            base: path,
+            path: _getDistFilePath(file, path, true),
+        });
+        mappings.basename = _getRevisedFileExtension(file.basename) + '.map';
+        const fileContent = result.outputText ?  new Buffer(`${result.outputText}\n\n//# sourceMappingURL=${result.sourceMapText}`) : new Buffer(result.outputText);
+        const distFile = _getDistFile(file, path, true, fileContent);
+        return [mappings, distFile, declarationFile];
+    })
+    .filter((a)=> !!a)
+    .reduce((a, b) => a.concat(b), [declarationFile])
+
+    return results
+}
 const _toCompile = (file) => {
-  return compiledFileTypes.indexOf(file.extname.replace('.', '')) > -1 && !file.stem.endsWith('.d');
+    return compiledFileTypes.indexOf(file.extname.replace('.','')) > -1;// && !file.stem.endsWith('.d');
```

Comment:
Are you sure about this change? bit will try to compile d.ts files.. which will makes problems

Reply:
changed